driver/ppm_events.c

```diff
@@ -249,14 +249,16 @@ inline u32 compute_snaplen(struct event_filler_arguments *args, char *buf, u32 l
 			if (err == 0) {
 				if(args->event_type == PPME_SOCKET_SENDTO_X)
 				{
+					unsigned long syscall_args[6] = {};
 					unsigned long val;
 					struct sockaddr __user * usrsockaddr;
 					/*
 					 * Get the address
 					 */
-					if (!args->is_socketcall)
-						syscall_get_arguments(current, args->regs, 4, 1, &val);
-					else
+					if (!args->is_socketcall) {
+						ppm_syscall_get_arguments(current, args->regs, syscall_args);
+						val = syscall_args[4];
+					} else
```

Comment:
This pattern repeats several times, can we do something like this somewhere up? hopefully even outside the `args->event_type` checks

```
unsigned long *actual_args = &syscall_args;

if (!args->is_socketcall) {
    ppm_syscall_get_arguments(current, args->regs, syscall_args);
} else {
    actual_args = &args->socketcall_args;
}
```

and unconditionally refer to `actual_args` (hopefully with a better name) everywhere?

Yeah, I know, it's been there already but since we're here... ;)

Reply:
It's unfortunately not as clean as your sample code makes it out to be. In the call site you refer to (inside compute_snaplen), val refers to different slots in syscall_args depending on some flags. So the code would actually look like:

`unsigned long *actual_args = &syscall_args[4];

if (!args->is_socketcall) {
    ppm_syscall_get_arguments(current, args->regs, syscall_args);
} else {
    actual_args = &args->socketcall_args[4];
}

if (usrsockaddr == NULL) {
   actual_args = &syscall_args[5];
   if (!args->is_socketcall) {
       ppm_syscall_get_arguments(current, args->regs, syscall_args);
   } else {
       actual_args = &args->socketcall_args[5];
   }
}
`
Since we have to shuffle the index around, I don't feel as though we get as much benefit from a change like that. I don't feel that it outweighs the awkwardness of having actual_args being a pointer to a long. I also feel like that version is a little less readable because actual_args gets set to the correct index much earlier in the function instead of right at the relevant call site.

Perhaps I'm misunderstanding your comment though?

Reply:
@nathan-b I had something like this in mind: https://github.com/draios/sysdig/commit/bed44dd7f5678647c955164e66934663ccaa97c8

(though it's obviously unfinished and there are `args->is_socketcall` checks in other files as well, but that's the general idea)

Reply:
Gotcha. I'm on the fence about whether this is future work or something to be resolved while I'm already touching this code, but since I'm not allowed to check in right now anyway then maybe tomorrow I'll look at doing this.

Reply:
yeah, no pressure, we can revisit this later and just do the minimal fix now